vignettes/retrieved_dropout.Rmd

```diff
@@ -0,0 +1,318 @@
+different types of ICEs. 
+
+Specifically, we simulate a 1:1 randomized trial of an active drug (intervention) versus placebo (control) with 100 subjects per 
+group and 4 post-baseline assessments (3-monthly visits until 12 months):
+
+- The mean outcome trajectory in the placebo group increases linearly from 50 at baseline (visit 0) to 60 at visit 4, i.e. the slope is 10 points/year.
+- The mean outcome trajectory in the intervention group is identical to the placebo group up to month 3. From month 3 onward, the slope decreases by 50% to 5 points/year.
+- The covariance structure of the baseline and follow-up values in both groups is implied by a random intercept and slope model 
+with a standard deviation of 5 for both the intercept and the slope, and a correlation of 0.25. In addition, an independent 
+residual error with standard deviation 2.5 is added to each assessment.   
+- The probability of the intercurrent event study drug discontinuation after each visit is calculated according to a logistic model which depends on the observed outcome at that visit.
+Specifically, a visit-wise discontinuation probability of 3% and 4% in the control and intervention group, respectively, is specified in case the observed outcome is equal to 50 (the mean value at baseline).
+The odds of a discontinuation is simulated to increase by +10% for each +1 point increase of the observed outcome.
+- Study drug discontinuation is simulated to have no effect on the mean trajectory in the placebo group.
+In the intervention group, subjects who discontinue follow the slope of the mean trajectory from the placebo group
+from that time point onward. This is compatible with a copy increments in reference (CIR) assumption. 
+- Study dropout at the study drug discontinuation visit occurs with a probability of 50% leading to missing outcome
+data from that time point onward.
+
+The function `simulate_data()` requires 3 arguments (see the function documentation `help(simulate_data)` for more details): 
+
+- `pars_c`: The simulation parameters of the control group.
+- `pars_t`: The simulation parameters of the intervention group.
+- `post_ice1_traj`: Specifies how observed outcomes after ICE1 are simulated.
+
+
+Below, we report how data according to the specifications above can be simulated with function `simulate_data()`:
+
+```{r}
+library(rbmi)
+library(dplyr)
+
+set.seed(1392)
+
+n <- 100
+time <- c(0, 3, 6, 9, 12)
+
+# Mean trajectory control
+muC <- c(50.0, 52.5, 55.0, 57.5, 60.0)
+
+# Mean trajectory intervention
+muT <- c(50.0, 52.5, 55.0, 56.25, 57.50)
+
+# Create Sigma
+sd_error <- 2.5
+covRE <- rbind(
+  c(25.0, 6.25),
+  c(6.25, 25.0)
+)
+
+Sigma <- cbind(1, time / 12) %*% covRE %*% rbind(1, time / 12) + diag(sd_error^2, nrow = length(time))
+
+# Set probability of discontinuation
+probDisc_C <- 0.03
+probDisc_T <- 0.04
+or_outcome <- 1.10 # +1 point increase => +10% odds of discontinuation
```

Comment:
I know it is not optimal from a programming perspective, I thought it would help with code readability. But if you want I can change it to set_simul_pars(prob_ice1 = 0.03 ,... ) instead of set_simul_pars(prob_ice1 = probDisc_C ,... ) ..

Reply:
Yes, I actually prefer to do as you say, I will also add comments to explain what those parameters refer to.